src/System.IO.FileSystem/src/System/IO/Directory.cs

```diff
@@ -563,6 +563,9 @@ public static void Move(String sourceDirName, String destDirName)
             String destinationRoot = Path.GetPathRoot(destPath);
             if (!String.Equals(sourceRoot, destinationRoot, pathComparison))
                 throw new IOException(SR.IO_SourceDestMustHaveSameRoot);
+            
+            if (FileSystem.Current.DirectoryExists(fulldestDirName))
+                throw new IOException(SR.Format(SR.IO_AlreadyExists_Name, fulldestDirName));
```

Comment:
> Also, what's the desired behavior if, say, a file exists at the specified name? Do we currently do the right thing and do we have tests for that? What about a symlink?

The MoveDirectory function translates to a call to `rename(2)`. When the source is a directory the destination must be non-existent or an empty directory. If the source is a symlink then the link will be renamed. It doesn't look like we have any tests for moving onto an existing file. I will add one.